File: src/Omnia.ProcessManagement.Models/ProcessTypes/PublishingApprovalSettings/TermDrivenPublishingApprovalSettings.cs
using Omnia.Fx.Models.Users;
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.ProcessTypes.PublishingApprovalSettings
{
    public class TermDrivenPublishingApprovalSettings : PublishingApprovalSettings
    {
        public override PublishingApprovalSettingsTypes Type
        {
            get
            {
                return PublishingApprovalSettingsTypes.TermDriven;
            }
        }

        public Guid TaxonomyEnterprisePropertyDefinitionId { get; set; }
        public Dictionary<Guid, List<UserIdentity>> Settings { get; set; }
    }
}

File: src/Omnia.ProcessManagement.Web/client/fx/models/data/processtypes/publishingapprovalsettings/TermDrivenPublishingApprovalSettings.ts
import { PublishingApprovalSettings, PublishingApprovalSettingsTypes } from '.';
import { GuidValue, UserIdentity } from '@omnia/fx-models';

export interface TermDrivenPublishingApprovalSettings extends PublishingApprovalSettings {
    type: PublishingApprovalSettingsTypes.TermDriven;

    taxonomyEnterprisePropertyDefinitionId: GuidValue;
    /**
     * key is term/termset id
     * let the dictionary value null to be inherit
     * 
     * */
    settings: { [id: string]: Array<UserIdentity> };
}
